### PR TITLE
build: Try using devenv.sh

### PR DESCRIPTION
**Pull request by jimmidyson**

### CODE REVIEW

**Comment by jimmidyson** [comment]
Oh wow 30 minutes initial install??? That's crazy. I wonder what took all that time. That is an awful experience.

**Comment by jimmidyson** [comment]
Interesting that it only takes about 3m30s in GHA on macos - see https://github.com/mesosphere/mindthegap/actions/runs/5191203933/jobs/9358720988?pr=428

**Comment by jimmidyson** [comment]
@dkoshkin i wonder if you could uninstall nix - perhaps following https://nixos.org/manual/nix/unstable/installation/uninstall.html#macos

and then retrying installation with

```
sh <(curl -L https://nixos.org/nix/install) --no-channel-add --daemon-user-count 6 --yes
```

**Comment by jimmidyson** [comment]
i added some more instructions for installation at https://github.com/mesosphere/mindthegap/pull/428/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R168-R215 which hopefully should be faster for you to get set up :crossed_fingers::skin-tone-4:

**Comment by dkoshkin** [comment]
Thanks for the suggestions @jimmidyson!
Now it took ~2 mins to install nix and 2-3 mins to install devenv.

The process seems to just work:
```
➜  mindthegap git:(main) git checkout jimmi/devenv
Switched to branch 'jimmi/devenv'
Your branch is up to date with 'origin/jimmi/devenv'.
WARNING: Multiple 'url.*..insteadof' keys with the same alias: "https://github.com/"
direnv: loading ~/src/github.com/mesosphere/mindthegap/.envrc
direnv: loading https://raw.githubusercontent.com/cachix/devenv/d1f7b48e35e6dee421cfd0f51481d17f77586997/direnvrc (sha256-YBzqskFZxmNb3kYVoKD9ZixoPXJh1C9ZvTLGFRkauZ0=)
direnv: using devenv
direnv: .envrc changed, reloading
Building shell ...
direnv: ([/usr/local/bin/direnv export zsh]) is taking a while to execute. Use CTRL-C to give up.
direnv: updated devenv shell cache
direnv: export +C_INCLUDE_PATH +DEVENV_DOTFILE +DEVENV_PROFILE +DEVENV_ROOT +DEVENV_STATE +GOPATH +GOROOT +IN_NIX_SHELL +LD_LIBRARY_PATH +LIBRARY_PATH +PKG_CONFIG_PATH +XDG_CONFIG_DIRS +XDG_DATA_DIRS +name ~PATH
➜  mindthegap git:(jimmi/devenv) make build-snapshot
▶ building snapshot
  • starting build...
  • loading config file                              file=.goreleaser.yml
  • building only for darwin/amd64                   reason=single target is enabled
  • loading environment variables
  • getting and validating git state
    • building...                                    commit=f32247916998bc548acc4b6c180dd87d43841dde latest tag=v1.7.7
    • pipe skipped                                   reason=disabled during snapshot mode
  • parsing tag
  • setting defaults
  • snapshotting
    • building snapshot...                           version=v1.8.0-dev
  • checking distribution directory
    • cleaning dist
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/mindthegap_darwin_amd64_v1/mindthegap
    • took: 5s
  • storing release metadata
    • writing                                        file=dist/artifacts.json
    • writing                                        file=dist/metadata.json
  • build succeeded after 5s
  • thanks for using goreleaser!
➜  mindthegap git:(jimmi/devenv)
```

File: test/e2e/imagebundle/push_image_bundle_test.go [hunk 1]
```diff
@@ -54,11 +55,16 @@ var _ = Describe("Push Bundle", func() {
 		registryScheme string,
 		registryInsecure bool,
 	) {
+		httpProxy := os.Getenv("http_proxy")
```
Comment (dkoshkin) [inline test/e2e/imagebundle/push_image_bundle_test.go:58]: Are these changes necessary in this PR?

**Review by dkoshkin — APPROVED** [review]
I don't have strong options on either of the tools, but if you prefer this one after working with both I support you in that.

Seems to work well enough, the only concern I had was the setup time and thats the uninstall is a little complicated.

File: test/e2e/imagebundle/push_image_bundle_test.go (hunk 1 above)
Comment (jimmidyson) [reply test/e2e/imagebundle/push_image_bundle_test.go]: No, moved into another PR.
